File: lib/qt4/ItemListComboBox.cpp
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <cstdint>
#include <string>
#include <vector>

#include <pkmn/lists.hpp>
#include <pkmn/database.hpp>
#include <pkmn/qt4/ItemListComboBox.hpp>

namespace pkmn
{
    namespace qt4
    {
        ItemListComboBox::ItemListComboBox(const unsigned int version_id, QWidget* parent):
            QComboBox(parent)
        {
            std::vector<pkstring> item_vec;
            get_item_list(item_vec, version_id);

            for(unsigned int i = 0; i < item_vec.size(); i++)
                addItem(QString::fromUtf16(item_vec[i]), QVariant(i));
        }
    } /* namespace qt4 */
} /* namespace pkmn */

File: lib/swig/java/ComboBoxes/ItemListComboBox.java
/*
 * Copyright (c) 2013-2015 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.*;

/**
 * A ComboBox that lists all item available in the given generation.
 *
 * @param gen highest generation whose items to include
 */
public class ItemListComboBox extends JComboBox<String>
{
    public ItemListComboBox(int gen)
    {
        super();

        StringVector itemVec = new StringVector();
        Lists.getItemList(itemVec, gen);
        for(int i = 0; i < itemVec.size(); i++)
        {
            addItem(itemVec.get(i));
        }

        setEditable(false);
    }
}
